lib/deb/s3/release.rb

```diff
@@ -31,10 +31,10 @@ def retrieve(codename, origin=nil, suite=nil, cache_control=nil)
         rel = self.parse_release(s)
       else
         rel = self.new
-        rel.codename = codename
-        rel.origin = origin unless origin.nil?
-        rel.suite = suite unless suite.nil?
       end
+      rel.codename = codename
+      rel.origin = origin unless origin.nil?
+      rel.suite = suite unless suite.nil?
```

Comment:
Unsure what the original intent here was, but this PR will not allow one to unset the origin or suite once set.  I don't know if that is a valid use-case or not.  If it is, let me know and it can be fixed (remove the `.nil?` check).